### PR TITLE
add kpi views and forecasting for mobile

### PR DESCRIPTION
**Pull request by irrationalagent**
This mirrors how things work for desktop and adds a similar KPI/forecasting setup for mobile. 

[Here is the query](
https://console.cloud.google.com/bigquery?sq=630180991450:622632e0fa804270bc844f9f271087ce
) that combines the existing "official" top-level forecasts into one table. We might consider moving this into the static dir of bigquery-etl

[Here is the query](
https://console.cloud.google.com/bigquery?sq=630180991450:3389b92628b04820abd7dfdedec75d7c
) that creates the `mozdata.analysis.mobile_dou_forecasts_cache` table and `mozdata.analysis.mobile_dou_forecasts` view.  I'll open a PR at bigquery ETL to codify it when we're ready.

I tested and this works locally but I need to add the target levels in, so marking as a draft until I do that.

### CODE REVIEW

**Comment by jklukas** [comment]
I copied the official forecasts table into `static` in shared-prod so we can reference that rather than `analysis` here:

```
› bq cp mozdata:analysis.mobile_forecasts_official moz-fx-data-shared-prod:static.mobile_forecasts_official_2021

Table 'mozdata:analysis.mobile_forecasts_official' successfully copied to 'moz-fx-data-shared-prod:static.mobile_forecasts_official_2021'
```

**Comment by irrationalagent** [comment]
@fbertsch does this look sane to you? if you think this works I will open a pr in bigquery-etl to codify the cache.

note i also changed the desktop forecast views to check and write to the official cache now. i think that's what we want right?

File: KPI/views/firefox_desktop_usage_forecast.view.lkml [hunk 1]
```diff
@@ -112,12 +112,12 @@ view: insert_stmnt {
...
       SET row_count = (
         SELECT COUNT(*)
-        FROM mozdata.analysis.dou_forecasts
+        FROM moz-fx-data-shared-prod.telemetry.looker_forecasts_cache
         WHERE key = current_key);
 
       IF row_count = 0 THEN
 
-      INSERT INTO mozdata.analysis.dou_forecasts_cache
+      INSERT INTO moz-fx-data-shared-prod.telemetry.looker_forecasts_cache
```
Comment (fbertsch) [inline KPI/views/firefox_desktop_usage_forecast.view.lkml:120]:
```suggestion
      INSERT INTO moz-fx-data-shared-prod.telemetry_derived.looker_forecasts_cache_v1
```

Comment (fbertsch) [reply KPI/views/firefox_desktop_usage_forecast.view.lkml:120]: I'm wondering if the service creds we use for this connection have write capabilities for shared-prod.

File: KPI/views/mobile_usage_2021.view.lkml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    suggest_explore: mobile_usage_fields
+    suggest_dimension: mobile_usage_fields.campaign
+    type: string
+  }
+
+  filter: canonical_app_name {
+    suggest_explore: mobile_usage_fields
+    suggest_dimension: mobile_usage_fields.canonical_app_name
+    type: string
+  }
```
Comment (fbertsch) [inline KPI/views/mobile_usage_2021.view.lkml:35]: This means we will have both a filter-only field `canonical_app_name`, and a dimension `app_name` in this explore. That may be confusing, we could drop the `app_name` dimension, make this a dimension, and keep the suggest; that will keep it as a filter in the above query but also let it be a dimension in the explore.

File: KPI/views/mobile_usage_fields.view.lkml [hunk 3]
```diff
@@ -0,0 +1,83 @@
+view: mobile_usage_fields {
+  derived_table: {
+    sql:
+      SELECT
+        * EXCEPT(app_name, canonical_app_name),
+        CASE WHEN app_name IN ('fennec', 'fenix') THEN 'fennec_fenix' ELSE app_name END AS app_name,
+        CASE WHEN canonical_app_name IN ('Firefox for Android (Fennec)', 'Firefox for Android (Fenix)') THEN 'Firefox for Android (Fennec + Fenix)' ELSE canonical_app_name END AS canonical_app_name
+      FROM `mozdata.telemetry.mobile_usage_2021` ;;
```
Comment (fbertsch) [inline KPI/views/mobile_usage_fields.view.lkml:8]: Nice, you nailed it

**Review by fbertsch — APPROVED** [review]
r+wc

File: KPI/views/firefox_desktop_usage_forecast.view.lkml [hunk 4]
```diff
@@ -68,7 +68,7 @@ view: new_profiles_model {
 
     SET row_count = (
       SELECT COUNT(*)
-      FROM mozdata.analysis.dou_forecasts
+      FROM moz-fx-data-shared-prod.telemetry.looker_forecasts_cache
```
Comment (fbertsch) [inline KPI/views/firefox_desktop_usage_forecast.view.lkml:71]:
```suggestion
      FROM mozdata.analysis.dou_forecasts
```

File: KPI/views/firefox_desktop_usage_forecast.view.lkml [hunk 5]
```diff
@@ -112,12 +112,12 @@ view: insert_stmnt {
 
       SET row_count = (
         SELECT COUNT(*)
-        FROM mozdata.analysis.dou_forecasts
+        FROM moz-fx-data-shared-prod.telemetry.looker_forecasts_cache
```
Comment (fbertsch) [inline KPI/views/firefox_desktop_usage_forecast.view.lkml:115]:
```suggestion
        FROM mozdata.analysis.dou_forecasts
```

File: KPI/views/firefox_desktop_usage_forecast.view.lkml [hunk 6]
```diff
@@ -112,12 +112,12 @@ view: insert_stmnt {
...
         WHERE key = current_key);
 
       IF row_count = 0 THEN
 
-      INSERT INTO mozdata.analysis.dou_forecasts_cache
+      INSERT INTO moz-fx-data-shared-prod.telemetry_derived.looker_forecasts_cache_v1
```
Comment (fbertsch) [inline KPI/views/firefox_desktop_usage_forecast.view.lkml:120]:
```suggestion
      INSERT INTO mozdata.analysis.dou_forecasts_cache
```

File: KPI/views/firefox_desktop_usage_forecast.view.lkml [hunk 7]
```diff
@@ -188,7 +188,7 @@ view: prediction {
     SELECT
       *
     FROM
-      mozdata.analysis.dou_forecasts
+      moz-fx-data-shared-prod.telemetry.looker_forecasts_cache
```
Comment (fbertsch) [inline KPI/views/firefox_desktop_usage_forecast.view.lkml:191]:
```suggestion
      mozdata.analysis.dou_forecasts
```

File: KPI/views/firefox_desktop_usage_forecast.view.lkml [hunk 8]
```diff
@@ -23,7 +23,7 @@ view: dau_model {
 
     SET row_count = (
       SELECT COUNT(*)
-      FROM mozdata.analysis.dou_forecasts
+      FROM moz-fx-data-shared-prod.telemetry.looker_forecasts_cache
```
Comment (fbertsch) [inline KPI/views/firefox_desktop_usage_forecast.view.lkml:26]:
```suggestion
      FROM mozdata.analysis.dou_forecasts
```

**Review by fbertsch — COMMENTED** [review]
Don't merge yet, we need to switch back to the

**Comment by fbertsch** [comment]
Okay, should be good now

**Pull request merged**